### PR TITLE
Request and Deploy Title Copy Link

### PR DESCRIPTION
**Pull request by Calvinp**
Adds a link next to the request ID to copy it to the clipboard.

TODO:
- [x] Make the link disappear on mouse off
- [x] Properly handle short request IDs

### CODE REVIEW

File: SingularityUI/app/components/requestDetail/header/RequestTitle.jsx [hunk 1]
```diff
@@ -16,40 +17,67 @@ const errorDescription = (requestAPI) => {
...
+          </span>
         </span>
-      </span>
+      );
+    }
+
+    const requestIdToDisplay = Utils.maybe(requestAPI, ['data', 'request', 'id']) || requestId;
+
+    return (
+      <div onMouseOver={() => this.onMouseOver()}>
```
Comment (zdhickman) [inline SingularityUI/app/components/requestDetail/header/RequestTitle.jsx]: If `this.onMouseOver` is bound, you can just do `onMouseOver={this.onMouseOver}`

**Comment by Calvinp** [comment]
Trying to have a link pop up on hover that could be clicked to copy things wasn't working. Instead, now the requestID itself can be clicked to copy it.

An overlay trigger explains this fact to the user when they hover over the request id.

**Comment by Calvinp** [comment]
@tpetr @kwm4385 @wolfd this PR is good to be reviewed/merged.

File: SingularityUI/app/components/requestDetail/header/RequestTitle.jsx [hunk 2]
```diff
@@ -39,13 +40,22 @@ const RequestTitle = ({requestId, requestAPI}) => {
     );
   }
 
+  const requestIdToDisplay = Utils.maybe(requestAPI, ['data', 'request', 'id']) || requestId;
```
Comment (tpetr) [inline SingularityUI/app/components/requestDetail/header/RequestTitle.jsx]: why bother with the `maybe()` if we have the `requestId` variable?

Comment (Calvinp) [inline SingularityUI/app/components/requestDetail/header/RequestTitle.jsx]: Honestly, I don't know. I copied it from what was there before (see line 48 on the other side of the diff), but I can't justify it.
I'll delete it - `requestId` is marked as required in propTypes after all.

**Comment by Calvinp** [comment]
I also added the same ability to the Deploy Detail page. It doesn't make sense to add it to the Task Detail page because the task ID isn't displayed as the title.

**Comment by tpetr** [comment]
LGTM

**Pull request merged**